lib/tree-view.coffee

```diff
@@ -671,6 +677,54 @@ class TreeView extends View
   toggleSide: ->
     toggleConfig('tree-view.showOnRightSide')
 
+  toggleShortcuts: ->
+    if @element.classList.contains 'shortcuts'
+      @disableShortcuts()
+    else
+      @enableShortcuts()
+
+  disableShortcuts: ->
+    @removeShortcuts()
+    @emitter.off 'did-toggle-directory', @updateAllTreeStartIndexes
+    @element.classList.remove 'shortcuts'
+
+  enableShortcuts: ->
+    @emitter.on 'did-toggle-directory', @updateAllTreeStartIndexes
+    @updateAllTreeStartIndexes()
+    @element.classList.add 'shortcuts'
+
+  updateAllTreeStartIndexes: =>
+    @removeShortcuts()
+    @setStartIndexesWithin 0, @element.querySelector('.tree-view-scroller')
+
+  setStartIndexesWithin: (index, el) ->
```

Comment:
Actually, this is a bonkers name. Perhaps something more along the lines of `addShortcutsToListTrees`?